File: Examples/VisualBasic/Miscellaneous/SortResourcesByName.vb

Namespace Miscellaneous
    Public Class SortResourcesByName
        Public Shared Sub Run()
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName)

            ' ExStart:SortResourcesByName
            Dim project As New Project(dataDir & "project-sort.mpp")

            Dim resources As List(Of Resource) = project.Resources.ToList()
            resources.Sort(New RscNameComparer())

            For Each rsc As Resource In resources
                Console.WriteLine(rsc)
            Next
            ' ExEnd:SortResourcesByName
        End Sub

        ' ExStart:ImplementIComparer
        Private Class RscNameComparer
            Implements IComparer(Of Resource)
            Public Function Compare(x As Resource, y As Resource) As Integer Implements IComparer(Of Aspose.Tasks.Resource).Compare
                If String.IsNullOrEmpty(x.[Get](Rsc.Name)) Then
                    Return 1
                End If
                If String.IsNullOrEmpty(y.[Get](Rsc.Name)) Then
                    Return -1
                End If
                Return x.[Get](Rsc.Name).CompareTo(y.[Get](Rsc.Name))
            End Function
        End Class
        '  ExEnd:ImplementIComparer
    End Class
End Namespace

File: Examples/CSharp/Miscellaneous/SortResourcesByName.cs
namespace Aspose.Tasks.Examples.CSharp.Miscellaneous
{
    using System;
    using System.Collections.Generic;

    internal class SortResourcesByName
    {
        private static readonly string DataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod());
        
        //ExStart:SortResourcesByName
        //ExFor: Project.Resources
        //ExSummary: Shows how to sort resources by name.
        public static void Run()
        {
            var project = new Project(DataDir + "project-sort.mpp");

            List<Resource> resources = project.Resources.ToList();
            resources.Sort(new RscNameComparer());

            foreach (var rsc in resources)
            {
                Console.WriteLine(rsc);
            }
        }

        private class RscNameComparer : IComparer<Resource>
        {
            public int Compare(Resource x, Resource y)
            {
                if (x == null && y == null)
                {
                    return 0;
                }

                if (x == null)
                {
                    return -1;
                }

                if (y == null)
                {
                    return 1;
                }

                if (string.IsNullOrEmpty(x.Get(Rsc.Name)))
                {
                    return 1;
                }

                if (string.IsNullOrEmpty(y.Get(Rsc.Name)))
                {
                    return -1;
                }
                
                return string.Compare(x.Get(Rsc.Name), y.Get(Rsc.Name), StringComparison.Ordinal);
            }
        }
        //ExEnd:SortResourcesByName
    }
}
